tests/subsys/audio_module/src/audio_module_test_common.c

```diff
@@ -96,7 +98,10 @@ int test_data_process_function(struct audio_module_handle_private *handle,
 	ARG_UNUSED(handle);
 
 	memcpy(audio_data_tx, audio_data_rx, sizeof(struct audio_data));
-	memcpy(audio_data_tx->data, audio_data_rx->data, audio_data_rx->data_size);
+
+	if (audio_data_rx->data != NULL && audio_data_tx->data != NULL) {
+		memcpy(audio_data_tx->data, audio_data_rx->data, audio_data_rx->data_size);
+	}
```

Comment:
Should there be a warn/fail if this is not true?

Reply:
It seems strange to write the `data_size` on the line below if there are NULL pointers present.

Reply:
Discuss. I need to understand the context here a bit better.